### DISCUSSION TITLE
Larger display that could be mounted in the Hat labs SH-ESP 32 J-Box

### DISCUSSION THREAD

**Discussion by RichFind** [hatlabs/discussions#77] [Q&A]
Any recommendations for a larger display? Not being much of a programmer, how hard it is to modify the software to work with a different display. What about the heat generated by the larger display?
TIA

**Comment by RichFind** [comment]
Probably should have mentioned the engine HAT is installed.

**Comment by wjquigs11** [comment]
Do you currently have a display connected, and if so, is it the 0.9" OLED? If it is, it's using the I2C protocol, so (in theory at least) you could replace it with any other I2C display. I recommend Adafruit, because they have great instructional blogs and a support forum, and I have one of these:
 https://learn.adafruit.com/1-5-and-2-4-monochrome-128x64-oled-display-module
However, most displays are configured for SPI, so you either have to solder/unsolder some jumpers on the back to switch to I2C, or use SPI. There's nothing difficult about SPI but it does require changing some code and connecting the right wires, but if you've already gotten the engine HAT connected, it's no more complex than that.
